packages/shared/components/modals/CollectibleDetailsMenu.svelte

```diff
@@ -1,15 +1,45 @@
 <script lang="typescript">
     import { Modal, MenuItem } from 'shared/components'
     import { localize } from '@core/i18n'
+    import { closePopup, openPopup } from '@auxiliary/popup/actions'
+    import { checkActiveProfileAuth } from '@core/profile/actions'
+    import { burnNft } from '@core/wallet'
+    import { INft } from '@core/nfts'
+    import { CollectiblesRoute, collectiblesRouter } from '@core/router'
 
     export let modal: Modal
+    export let nft: INft
+
+    function openBurnNft(): void {
+        openPopup({
+            type: 'confirmation',
+            props: {
+                title: localize('actions.confirmNftBurn.title', {
+                    values: {
+                        nftName: nft.name,
+                    },
+                }),
+                description: localize('actions.confirmNftBurn.description'),
+                hint: localize('actions.confirmNftBurn.hint'),
+                warning: true,
+                confirmText: localize('actions.burnToken'),
+                onConfirm: () => {
+                    checkActiveProfileAuth(async () => {
+                        await burnNft(nft.id)
```

Comment:
Potentially a good idea to put a Try (burn and navigate) Catch (burn error) Finally (close popup) statement here in case the burnNft throws an error from wallet.rs. That way we aren't navigating to gallery view if an error is thrown and the Nft isn't deleted

Reply:
The potential errors are already included in the `burnNft` function

Reply:
What I mean is that if the burnNft call throws an error we still end up routing away back to the CollectiblesGalleryView. I think if there is an error than we should stay in the DetailsView.